lib/facter/custom_facts/core/execution/base.rb

```diff
@@ -77,12 +79,31 @@ def builtin_command?(command)
           output.chomp =~ /builtin/ ? true : false
         end
 
-        def execute_command(command, on_fail, logger = nil)
+        def execute_command(command, on_fail, logger = nil, time_limit = nil)
+          @log ||= Log.new(self)
```

Comment:
shouldn't the log be initialised outside the method?

Reply:
Done